src/OrleansPSUtils/StartGrainClient.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Orleans;
+using Orleans.Runtime.Configuration;
+using System;
+using System.IO;
+using System.Management.Automation;
+using System.Net;
+
+namespace OrleansPSUtils
+{
+    [Cmdlet(VerbsLifecycle.Start, "GrainClient", DefaultParameterSetName = DefaultSet)]
+    public class StartGrainClient : PSCmdlet
+    {
+        private const string DefaultSet = "Default";
+        private const string FilePathSet = "FilePath";
+        private const string FileSet = "File";
+        private const string ConfigSet = "Config";
+        private const string EndpointSet = "Endpoint";
+
+        [Parameter(Position = 1, Mandatory = true, ValueFromPipeline = true, ParameterSetName = FilePathSet)]
+        public string ConfigFilePath { get; set; }
+
+        [Parameter(Position = 2, Mandatory = true, ValueFromPipeline = true, ParameterSetName = FileSet)]
+        public FileInfo ConfigFile { get; set; }
+
+        [Parameter(Position = 3, Mandatory = true, ValueFromPipeline = true, ParameterSetName = ConfigSet)]
+        public ClientConfiguration Config { get; set; }
+
+        [Parameter(Position = 4, Mandatory = true, ValueFromPipeline = true, ParameterSetName = EndpointSet)]
+        public IPEndPoint GatewayAddress { get; set; }
```

Comment:
ah, I see. Still, shouldn't you restart from position 1 for each of the sets? (completely naive question, I'm not familiar with it)